### PR TITLE
feat: add pip install pycasbin

### LINKED ISSUE #174: [PyCasbin] Make the library pip installable
From: https://github.com/casbin/casbin-cpp/issues/173

### Description

The current `pycabsin` target (library) can only be installed by [compiling and installing locally](https://github.com/casbin/casbin-cpp/tree/master/bindings#installing-the-pycasbin-module). People use Python for its utility, so we must make it simpler for the clients to install and use pycasbin (on cpp) with superior performance.

For reference, have a look at [tensorflow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python).

### PR DESCRIPTION
**Pull request by sheny1xuan**
Signed-off-by: stonex <1479765922@qq.com>

### Description
+ Make the library pip installable.
1. Add `setup.py` to build pycasbin library with pip. Unlike tensorflow, they build library with `Bazel`. We use cmake to build our python module.
2. Add `pycasbin` dir to store `python bind` dir. Remove `other language` bind Cmake choice, I think we won't use this file for a long while. 
3. In `pycasbin` Cmake file, I change it ref the [pybind11 cmake build guide](https://pybind11.readthedocs.io/en/stable/compiling.html#advanced-interface-library-targets). And I meet a strange thing, once we `find_package(python)` before `find_package(pybind11)`, won't pass `pybind ci` in `windows and macos`. I check the build `output`, find it will be generate conflict python module version. In this [action](https://github.com/sheny1xuan/casbin-cpp/runs/4379284647?check_suite_focus=true), I use `python3.8 in macos` for test, and it generate `python 3.9 module`. Anyway, when I remove `find_package(python)`, it works well.
image

4. Add python binding ci for all three platform like [pybind11 cmake_example](https://github.com/pybind/cmake_example).
5. Change a build choice in `ci`, because pybind11 will generate a [link error](https://github.com/sheny1xuan/casbin-cpp/runs/4365575090?check_suite_focus=true). And I find same issue in this [link](https://github.com/BlueBrain/nmodl/issues/266). Use `release` build choice to build pybind will avoid this error.
6. The last confused thing is, when I `add_subdirectory(pycasbin)` before `find_package(pybind11)`, I can't use the `cmake function` that pybind11 support. So, I move the `add_subdirectory(pycasbin)` after `find_package`, It works. 
7. All in all, I am not familiar with `cmake`. Although this work can pass the `python binding ci`, but maybe have something not good in the code. And I cost so much time running actions to fix ci in python binding and find the problem `3` :hushed:.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@EmperorYP7 plz review

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+# The information here can also be placed in setup.cfg - better separation of
+# logic and declaration, and simpler if you include description/version in a file.
+setup(
+    name="pycasbin",
+    version="0.0.0",
+    description="An authorization library that supports access control models like ACL, RBAC, ABAC in Python",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/casbin/pycasbin",
```
[the remark this answers is not in the document]
Comment (hsluoyz) [reply setup.py:129]: @sheny1xuan

[the remark this answers is not in the document]
Comment (sheny1xuan) [reply setup.py]: Fixed.

File: pycasbin/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    [pycasbin]: Build "pycasbin" target for Python Bindings
+    [pycasbin]: Add "lib/python3.9/site-packages" to your sys.path/USER_SITE variable if not already present
+    ```
+
+3. Build the python bindings (`pycasbin` target):
+    ```bash
+    cmake --build . --config Release --target pycasbin
+    ```
+
+4. Install the `pycasbin` module:
```
Comment (EmperorYP7) [inline pycasbin/README.md:36]: We also need to update the docs for installing through pip. Should we do it in this PR itself? @hsluoyz

Comment (hsluoyz) [reply pycasbin/README.md:36]: Sure @sheny1xuan

**Review by EmperorYP7 — APPROVED** [review]
LGTM

**Pull request merged**